promql/engine.go

```diff
@@ -576,26 +576,57 @@ func (ng *Engine) execEvalStmt(ctx context.Context, query *query, s *parser.Eval
 	return mat, warnings, nil
 }
 
-// subqueryOffsetRange returns the sum of offsets and ranges of all subqueries in the path.
-func (ng *Engine) subqueryOffsetRange(path []parser.Node) (time.Duration, time.Duration) {
+// subqueryOffsetRangeTimestamp returns the sum of offsets and ranges of all subqueries in the path.
+// It also returns the min and max timestamp associated with the subquery.
+func (ng *Engine) subqueryOffsetRangeTimestamp(path []parser.Node) (time.Duration, time.Duration, int64, int64) {
```

Comment:
That's a bit of a mouthful. Something like subqueryTimes would make more sense as this stage I think.